first-analysis-steps/analysisflow.md

```diff
@@ -0,0 +1,102 @@
+
+{% endcallout %}
+
+Discussions on the new analysis tools that might be useful for the LHCb community are held in the [work package 4](https://lhcb-dpa.web.cern.ch/lhcb-dpa/wp4/index.html) of the Data Processing and Analysis project (DPA). 
+
+### Analysis Preservation
+
+After one gets the necessary samples and writes the necessary macros and scripts to perform the analysis steps, like applying additional selections, fitting distributions, computing efficiencies and acceptances, etc. 
+After acquiring results, one has to have their work being review by physics working group (PWG) and then by collaboration. 
+
+One of the requirements for the successful review is that your analysis is fully preserved.  
+Scientific integrity means having transparency on each step of the research as well as reproducibility of the results.
+Both LHCb and CERN have guidelines which one should always follow when collecting research usign LHCb or CERN facilities.
+LHCb guidelines are discussed and implemented in the [work package 6](https://lhcb-dpa.web.cern.ch/lhcb-dpa/wp6/index.html) of DPA.
+CERN guideline can be found [here](https://analysispreservation.cern.ch/login?next=/).
+
+First step is to document well-used procedures, techniques and meticulously report the results of the analysis.
+This is done by writing a clear and extensive Analysis Note, that is then available to the collaborators.
+
+Secondly, all your code should have a [gitlab](https://about.gitlab.com/) repository.
+Gitlab is useful for tracing your own changes and changes of your collaborators too, as well as, tracing possible bugs with gitlab-CI.
+A lesson on gitlab can be found [here](https://hsf-training.github.io/analysis-essentials/git/README.html).
+
+However, it is not enough to just submit your code. 
+It is important to write a set of instructions to execute the steps in the correct sequence.
+You will quickly find yourself, that remembering how to run your own code even a couple of month after it was used for the last time can be really hard. 
+Instructions can be written using [snakemake](https://snakemake.readthedocs.io/en/stable/) and supportive bash scripts.
+
+Snakemake workflow management system allows you to write a pipeline for analysis that will automatically detect the hierarchy of the steps and changes in inputs (both samples and code). 
+One should aim to start implementing the snakemake workflow from the beginning of analysis development.
+Your snakemake workflow then should be committed to the gitlab repository with the final version of the code.
+
+The samples you have produced should also be saved.
+Analysis Productions help you to save a tested version of the DaVinci option files, that will help you to reproduce the ntuples with the same software version, that was specified in the original option files. 
+
+The resulting samples have to be uploaded to your working group eos. 
+
+You should always preserve the package versions of the sofrware that you have being using.
```

Comment:
```suggestion
You should always preserve the package versions of the software that you have been using.
```

Reply:
I am not sure this sends a clear message, but let the others comment. It would be good to say that _documenting_ your software versions in your analysis note is essential, but let's say I see little point in creating your own docker container just to store ROOT 6/24 which will anyway be accessible in other means.